extension-manifest-v3/src/store/account.js

```diff
@@ -11,36 +11,61 @@
 
 import { store } from 'hybrids';
 
-const API_URL = 'https://accountapi.ghostery.com/api/v2.1.0';
+const ACCOUNT_URL = 'https://accountapi.ghostery.com/api/v2.1.0';
+const AUTH_URL = 'https://consumerapi.ghostery.com/api/v2';
+const COOKIE_URL = 'https://ghostery.com';
 
-async function fetchApi(url, options) {
-  const csrfToken = await getCookie('csrf_token');
+const ALARM_NAME = 'account:refresh';
+const REFRESH_RATE = 60 * 24 * 30; // 30 days in minutes
 
-  const res = await fetch(`${API_URL}/${url}`, {
+async function cookie(name) {
+  // TODO: Add firstPartyDomain support
+  // firstPartyDomain: 'ghostery.com',
+  const cookie = await chrome.cookies.get({ url: COOKIE_URL, name });
+  return cookie ? cookie.value : null;
+}
+
+async function get(url) {
+  const accessToken = await cookie('access_token');
+  const csrfToken = await cookie('csrf_token');
+
+  if (!accessToken || !csrfToken) throw Error('Unauthorized');
+
+  const res = await fetch(url, {
     headers: {
       'Content-Type': 'application/vnd.api+json',
+      Authorization: `Bearer ${accessToken}`,
       'X-CSRF-Token': csrfToken,
     },
-    credentials: 'include',
-    ...options,
+    credentials: 'omit',
   });
 
   if (res.ok) {
     return res.json();
   }
 
-  throw Error(res.statusText);
+  throw res;
 }
 
-async function getCookie(name) {
-  const cookie = await chrome.cookies.get({
-    url: 'https://ghostery.com',
-    name,
-    // TODO: add firstPartyDomain support
-    // firstPartyDomain: 'ghostery.com',
-  });
+async function session(account) {
+  const userId = await cookie('user_id');
+  if (!userId) return undefined;
+
+  if (account && !(await cookie('access_token'))) {
+    const refreshToken = await cookie('refresh_token');
+    if (refreshToken) {
+      await fetch(`${AUTH_URL}/refresh_token`, {
```

Comment:
As a side-effect, this call updates cookies (access_token & refresh_token), right? Maybe we should still throw if the request fails (non 2xx).

Side-note: On Manifest V2, we changed the API to return also in the response body (for Firefox). As I understand, that workaround could now also be avoided, since we were forced to use httpOnly=false cookies. (But we would need to change the code there and also wait until old clients go away.)

Reply:
Your comment made me think about setting and clearing cookies, and as we omit credentials (we have to for Safari and upcoming Firefox), there must be logic for processing them. I have started work, but unfortunately, I'll not finish it today. Stay tuned on Monday.

Reply:
Should be ok now.